File: archives/python_legacy/backend/app/models/task.py
from pydantic import BaseModel
from typing import Optional, List

class TaskBase(BaseModel):
    title: str
    description: Optional[str] = None
    assigned_to: Optional[str] = None
    status: str = "To Do"  # To Do, In Progress, Done
    priority: str = "Medium"  # Low, Medium, High
    due_date: Optional[str] = None

class TaskCreate(TaskBase):
    pass

class TaskUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    assigned_to: Optional[str] = None
    status: Optional[str] = None
    priority: Optional[str] = None
    due_date: Optional[str] = None

class Task(TaskBase):
    id: str

File: backend/models/Task.js
const mongoose = require('mongoose');

const taskSchema = mongoose.Schema({
  tenantId: {
    type: String,
    required: true,
    default: 'default_tenant'
  },
  title: {
    type: String,
    required: true
  },
  description: {
    type: String,
    default: null
  },
  assigned_to: {
    type: String,
    default: null
  },
  status: {
    type: String,
    default: "To Do"
  },
  priority: {
    type: String,
    default: "Medium"
  },
  due_date: {
    type: String,
    default: null
  }
}, {
  collection: 'tasks_collection',
  timestamps: true
});

taskSchema.set('toJSON', {
  virtuals: true,
  versionKey: false,
  transform: function (doc, ret) {
    ret.id = ret._id.toString();
    delete ret._id;
  }
});

module.exports = mongoose.model('Task', taskSchema);
